qiskit_acqua/ising/graphpartition.py

```diff
@@ -0,0 +1,210 @@
+    Hamiltonian:
+    H = H_A + H_B
+    H_A = sum\_{(i,j)\in E}{(1-ZiZj)/2}
+    H_B = (sum_{i}{Zi})^2 = sum_{i}{Zi^2}+sum_{i!=j}{ZiZj}
+    H_A is for achieving goal 2 and H_B is for achieving goal 1.
+    """
+    num_nodes = len(weight_matrix)
+    pauli_list = []
+    shift = 0
+
+    for i in range(num_nodes):
+        for j in range(i):
+            if (weight_matrix[i,j] != 0):
+                wp = np.zeros(num_nodes)
+                vp = np.zeros(num_nodes)
+                vp[i] = 1
+                vp[j] = 1
+                pauli_list.append([-0.5, Pauli(vp, wp)])
+                shift += 0.5
+
+    for i in range(num_nodes):
+        for j in range(num_nodes):
+            if i != j:
+                wp = np.zeros(num_nodes)
+                vp = np.zeros(num_nodes)
+                vp[i] = 1
+                vp[j] = 1
+                pauli_list.append([1, Pauli(vp, wp)])
+            else:
+                shift += 1
+    return Operator(paulis=pauli_list), shift
+
+
+def parse_gset_format(filename):
+    """Read graph in Gset format from file.
+
+    Args:
+        filename (str): name of the file.
+
+    Returns:
+        numpy.ndarray: adjacency matrix as a 2D numpy array.
+    """
+    n = -1
+    with open(filename) as infile:
+        header = True
+        m = -1
+        count = 0
+        for line in infile:
+            v = map(lambda e: int(e), line.split())
+            if header:
+                n, m = v
+                w = np.zeros((n, n))
+                header = False
+            else:
+                s, t, x = v
+                s -= 1  # adjust 1-index
+                t -= 1  # ditto
+                w[s, t] = t
+                count += 1
+        assert m == count
+    w += w.T
+    return w
+
+def objective_value(x, w):
+    """Compute the value of a cut.
+
+    Args:
+        x (numpy.ndarray): binary string as numpy array.
+        w (numpy.ndarray): adjacency matrix.
+
+    Returns:
+        float: value of the cut.
+    """
+    X = np.outer(x, (1-x))
+    w_01 = np.where(w !=0, 1, 0)
+    print(w_01)
+    print(X)
+    print(w_01 * X)
+
+    return np.sum(w_01 * X)
+
+def get_graph_solution(x):
+    """Get graph solution from binary string.
+
+    Args:
+        x (numpy.ndarray) : binary string as numpy array.
+
+    Returns:
+        numpy.ndarray: graph solution as binary numpy array.
+    """
+    return 1 - x
+
+def sample_most_likely(n, state_vector):
+    """Compute the most likely binary string from state vector.
+    Args:
+        state_vector (numpy.ndarray or dict): state vector or counts.
+    Returns:
+        numpy.ndarray: binary string as numpy.ndarray of ints.
+    """
+    if isinstance(state_vector, dict) or isinstance(state_vector, OrderedDict):
+        # get the binary string with the largest count
+        binary_string = sorted(state_vector.items(), key=lambda kv: kv[1])[-1][0]
+        x = np.asarray([int(y) for y in list(binary_string)])
```

Comment:
@liupibm @chunfuchen this might need to be updated to `x = np.asarray([int(y) for y in reversed(list(binary_string))])`

Reply:
in my example, the code follows the else branch which properly reverses the bit string.
I am not sure about the if branch though. I guess sorted() will rank the qubits in the order [0,1,2,3]. @chunfuchen Can you check the if branch?

Reply:
the `sorted` sorts the all binary string by its counts, it did not change the qubit order; hence, the qubit order is still [N-1, N-2, ..., 0].